care/facility/api/serializers/patient_consultation.py

```diff
@@ -143,6 +148,20 @@ def validate_bed_number(self, bed_number):
             bed_number = None
         return bed_number
 
+    def validate_verified_by(self, verified_by):
+        if not verified_by.user_type == User.TYPE_VALUE_MAP["Doctor"]:
+            raise ValidationError("Only Doctors can verify a Consultation")
+
+        if (
+            verified_by.home_facility
+            and str(verified_by.home_facility.external_id)
+            != self.context["request"].data["facility"]
+        ):
+            raise ValidationError(
+                "Home Facility of the Doctor must be the same as the Consultation Facility"
+            )
```

Comment:
this validation should be moved to validate function and use consultation.facility instead of directly using the external_id

Reply:
how to get the consultation in order to access `consultaiton.facility`, when self.instance is not yet present?

Reply:
oh got it, we can add consultation in `get_serializer_context`